src/test/java/ua/lviv/iot/parkingServer/grpcservice/VehicleGrpcServiceTest.kt

```diff
@@ -0,0 +1,121 @@
+package ua.lviv.iot.parkingServer.grpcservice
+
+import com.example.VehicleGrpcServiceGrpc
+import com.example.VehicleGrpcServiceGrpc.VehicleGrpcServiceBlockingStub
+import com.example.VehicleOuterClass
+import io.grpc.ManagedChannel
+import io.grpc.ManagedChannelBuilder
+import java.time.Duration
+import org.assertj.core.api.Assertions
+import org.junit.jupiter.api.Assertions.assertEquals
+import org.junit.jupiter.api.BeforeEach
+import org.junit.jupiter.api.Test
+import org.springframework.beans.factory.annotation.Autowired
+import org.springframework.beans.factory.annotation.Value
+import org.springframework.boot.test.context.SpringBootTest
+import ua.lviv.iot.parkingServer.converter.VehicleConverter
+import ua.lviv.iot.parkingServer.model.Vehicle
+import ua.lviv.iot.parkingServer.model.enums.VehicleType
+import ua.lviv.iot.parkingServer.repository.VehicleRepository
+
+@SpringBootTest
+class VehicleGrpcServiceTest(
+    @Value("\${grpc.server.port}")
+    var grpcPort: Int
+) {
+
+    @Autowired
+    private lateinit var vehicleConverter: VehicleConverter
+
+    @Autowired
+    private lateinit var vehicleRepository: VehicleRepository
+
+    private lateinit var stub: VehicleGrpcServiceBlockingStub
+
+    private lateinit var channel: ManagedChannel
+
+    @BeforeEach
+    fun start() {
+        channel = ManagedChannelBuilder
+            .forTarget("localhost:$grpcPort")
+            .usePlaintext()
+            .build()
+        stub = VehicleGrpcServiceGrpc.newBlockingStub(channel)
+    }
+
+    @Test
+    fun createVehicle() {
+        val vehicle = Vehicle(
+            number = "KA6706VN",
+            typeOfVehicle = VehicleType.BUS,
+            durationOfUseOfParkingSpot = Duration.ofSeconds(3600),
+            isTicketReceived = true
+        )
```

Comment:
you recreate this object in every test method, maybe you can somehow get rid of repeating code?

Reply:
✅